File: apps/backend/src/main/java/tabletop/gather/backend/plan/UpdatePlanDto.java
package tabletop.gather.backend.plan;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import java.util.UUID;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.validator.constraints.Length;

@Getter
@Setter
public class UpdatePlanDto {

  @NotEmpty(message = "Plan name is required")
  @Length(max = 255, message = "Plan name cannot be longer than 255 characters")
  private String name;

  @NotNull(message = "Plan private is required")
  private Boolean isPrivate;

  @Length(max = 4000, message = "Plan description cannot be longer than 4000 characters")
  private String description;

  private int playerLimit;

  private UUID game;
}

File: apps/frontend/src/app/models/update-plan.dto.ts
import { Model, Uuid } from '../utils/types';
import { Dto } from './dto.base';

/**
 * Dto for creating a plan
 *
 * @param {string} name - The name of the plan
 * @param {boolean} isPrivate - Whether the plan is private
 * @param {string} description - The description of the plan
 * @param {number} playerLimit - The player limit of the plan
 * @param {Uuid} game - The game of the plan
 */
export class UpdatePlanDto extends Dto {
  public name!: string;
  public isPrivate!: boolean;
  public description!: string;
  public playerLimit!: number;
  public game?: Uuid;
}

export type UpdatePlan = Model<UpdatePlanDto>;
